packages/mui-material-nextjs/src/v13-appRouter/Link.tsx

```diff
@@ -0,0 +1,63 @@
+'use client';
+import * as React from 'react';
+import clsx from 'clsx';
+import { usePathname } from 'next/navigation';
+import NextLink, { LinkProps as NextLinkProps } from 'next/link';
+import MuiLink, { LinkOwnProps as MuiLinkProps } from '@mui/material/Link';
+import linkClasses from './linkClasses';
+
+export interface LinkProps
+  extends Omit<NextLinkProps, 'passHref' | 'legacyBehavior'>,
+    MuiLinkProps {
+  /**
+   * Extra class name to apply to the link.
+   */
+  className?: string;
+  /**
+   * The active class name to apply when the current page is the same as the link's href.
+   * @default 'Mui-active'
+   */
+  activeClassName?: string;
+}
+
+const Link = React.forwardRef<HTMLAnchorElement, LinkProps>(function Link(
+  {
+    href,
+    replace,
+    scroll,
+    shallow,
+    prefetch,
+    locale,
+    as,
+    className: classNameProp,
+    activeClassName = linkClasses.active,
+    ...muiLinkProps
+  },
+  ref,
+) {
+  const nextPathname = usePathname();
+  const pathname = typeof href === 'string' ? href : href.pathname;
+  const className = clsx(classNameProp, {
+    [activeClassName]: nextPathname === pathname && activeClassName,
+  });
+  return (
+    <NextLink
+      {...{
+        href,
+        replace,
+        scroll,
+        shallow,
+        prefetch,
+        locale,
+        as,
+      }}
+      // below props are required for NextLink to work with MUI Link
+      passHref
+      legacyBehavior
```

Comment:
Maybe we should see if we can avoid this?

Reply:
I don't think so based on Next.js [docs](https://nextjs.org/docs/pages/api-reference/components/link#if-the-child-is-a-functional-component).